src/shared/watchdog.c

```diff
@@ -186,4 +189,8 @@ void watchdog_close(bool disarm) {
         }
 
         watchdog_fd = safe_close(watchdog_fd);
+
+        /* Once closed, pinging the device becomes a NOP and we request a new
+         * call to watchdog_set_timeout() to open the device again. */
+        watchdog_timeout = USEC_INFINITY;
```

Comment:
if we go this way, then watchdog_set_timeout() should be renamed to watchdog_start() (or watchdog_setup(), or watchdog_prepare()) or so (since it suddenly is not just a boring setter, but the "constructor", that *must* be called, for anything to actually happen), but still take the timeout arg.

Reply:
Also, this commit as it stands now will break shutdown.c, since we don't call watchdog_set_timeout() there at all.

Reply:
good points